Source/Core/Common/HttpRequest.cpp

```diff
@@ -190,10 +209,24 @@ static size_t CurlWriteCallback(char* data, size_t size, size_t nmemb, void* use
   return actual_size;
 }
 
+static size_t header_callback(char* buffer, size_t size, size_t nitems, void* userdata)
+{
+  auto* headers = static_cast<HttpRequest::Headers*>(userdata);
+  const std::vector<std::string> key_value = SplitString(buffer, ':');
+  if (key_value.size() < 2)
+    return nitems * size;
+
+  const std::string value =
+      std::accumulate(key_value.begin() + 1, key_value.end(), std::string{}).erase(0, 1);
```

Comment:
Besides, I believe this would also strip out any `:` that follows the header name (as part of the value). At least I don't see any compensating `:` in the `std::accumulate` part that would add them back.